bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/PageImpl.java

```diff
@@ -166,4 +187,18 @@ public boolean hasCloudconfigSupport() {
         }
         return hasCloudconfigSupport;
     }
+
+    protected void filterClientlibCategories() {
```

Comment:
The logic here seems a bit over-complicated. What I would do instead is:
1. add all elements of `clientLibCategories` to a `LinkedHashSet`
2. iterate over the `headClientLibCategories` (no need to convert it to `List`, an array can be iterated as well)
3. remove from the set all matching head client libraries
4. replace the `clientLibCategories` with the contents of the set

Reply:
Fixed in 456d33fb04a9d21